back-end/src/main/java/project/UpdateRequest.java

```diff
@@ -0,0 +1,77 @@
+package project;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/** Created by wenbo on 1/4/18. */
+public class UpdateRequest {
+
+    public UpdateRequest() {}
+
+    // private variables
```

Comment:
@peterg17 can you help clean up the code a little bit to remove comments and logs for debugging? You can leave a few commented logs there if you think they are helpful for future debugging purposes.

Reply:
Yes, I can definitely clean up the comments. Thanks for pointing this out, comments and logging should not be in the committed code unless its very helpful for debugging.

Reply:
I went through and cleaned up comments/logging in the files I edited or added, mostly in UpdateRequestHandler.java, UpdateRequest.java, and jump.js. I also re-ran the election USMap visualization and nba visualization to make sure they still worked fine.

Reply:
This is great. Thanks! Let me know when the readme is ready. 

Also as a heads up - it might take a little more work before we can merge the PR. I looked through the code and there is a bunch places that I'd like to improve, including the API and where some code should live. But it should all be straightforward code refactoring. I will help - let's work on this together.

Reply:
I wasn't sure whether to update the README or the wiki, so I updated the Kyrix API wiki page for now. I noted my changes to the Transform object in the wiki here (https://github.com/tracyhenry/Kyrix/wiki/API-Reference#transform-for-updatable-layer). Also, I'm working on a new page in the wiki to describe updatable visualizations in a more end-to-end way that gives context. I added a new page in the wiki here (https://github.com/tracyhenry/Kyrix/wiki/Updatable-Kyrix-Visualizations), which I am adding content to. 

Thanks for the heads up, that makes total sense! It's a large PR and I would happy to help where I can to improve the code. We can coordinate here or over email, whatever works for you 👍

Reply:
Yes it's fine to put documentation in the places you mentioned. For the transform documentation, is it possible to add that to the existing parameter list? And as you mentioned in the comment, it'll be helpful to just 

In any case, I'll do one pass of review today. And let's communicate on Github here since it's easier  to reference code.

Reply:
Ok, i'll change the edits I made to the API Reference wiki page!

And just to clarify, only attributes that are listed in the schema object are made updatable.  The developer only needs to specify the identity reverse function lots of times if they want to allow the user to edit all of those attributes and only needs to be done once. I'm not sure how much effort is being saved by making a default function since they would still have to import and specify that default function in the Transform.

Reply:
Yes letting the user import `identityFunction` makes perfect sense. One improvement would be just asking the user to specify updatable attributes, and default unspecified ones to non-updatable. Then in your example we don't need to specify null for attributes such as `home_team` and `away_team`.